sha2/src/sha256/aarch64.rs

```diff
@@ -1,18 +1,24 @@
-use libc::{getauxval, AT_HWCAP, HWCAP_SHA2};
-
+#[cfg(target_os = "linux")]
 #[inline(always)]
 pub fn sha2_supported() -> bool {
+    use libc::{getauxval, AT_HWCAP, HWCAP_SHA2};
+
     let hwcaps: u64 = unsafe { getauxval(AT_HWCAP) };
     (hwcaps & HWCAP_SHA2) != 0
 }
 
+#[cfg(target_os = "macos")]
+#[inline(always)]
+pub fn sha2_supported() -> bool {
+    // TODO: Verify this is true for all available chips.
```

Comment:
It is. See the discussion here:

https://github.com/RustCrypto/utils/issues/378#issuecomment-826985574